src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1497,7 +1501,7 @@ primitive_inst::primitive_inst(network & network, program_node const& node, bool
     , _can_be_optimized(node.can_be_optimized())
     , _can_share_buffer(node.can_share_buffer())
     , _is_constant(node.is_constant())
-    , _needs_completion_event(is_any_user_cpu(node.get_users()) || node.is_output()) {
+    , _needs_completion_event(is_any_user_cpu(node.get_users()) || node.is_output() || has_runtime_skippable_users(node.get_users())) {
```

Comment:
will just check runtime skippable bring some extra sync where it was actually not needed？because some nodes marked as skippable may actually not able to be skipped during runtime?

Reply:
@ahnyoung-paul  What we need to check is not just "has_runtime_skippable" only but runtime_skiplable_user->runtime_skippable_user->....->optimized_out_user->cpu_user.  Current impl is adding sync point too much.

Reply:
@songbell Checking nested users skippability at runtime is also time consuming, so I think having conservative sync event is better. Of course 1) we need to reduce the scope as much as possible 2) need to check performance

Reply:
Fixed it.